### ISSUE TITLE
how to encrypt xls files? 

### ISSUE THREAD

**Issue by zurmokeeper** [zurmokeeper/officecrypto-tool#30]
how to encrypt xls files?

**Comment by zurmokeeper** [comment]
WX-   18826237239

**Comment by Lxxxxxxy** [comment]
> WX- 18826237239

1

**Comment by zurmokeeper** [comment]
> > WX- 18826237239
> 
> 1

可以加了，麻烦了

**Comment by cool-firer** [comment]
can not open `.xls `file after encryption. is this a bug?

**Comment by zurmokeeper** [comment]
> can not open `.xls `file after encryption. is this a bug?

@cool-firer Can you submit the original xls file and the encrypted file?
Let's check here.

**Comment by cool-firer** [comment]
Sure, files as below, and my code is like this:
```
//
const fs = require("fs");
const path = require("node:path");
const officeCrypto = require("officecrypto-tool");

function main() {
  const src = "./a/test1.xls";
  const input = fs.readFileSync(src);
  const output = officeCrypto.encrypt(input, { password: "123456" });

  const dst = "./a/test1_enc.xls";
  fs.writeFileSync(dst, output);
}

main();

```

[test1.xls](https://github.com/user-attachments/files/19935002/test1.xls)
[test1_enc.xls](https://github.com/user-attachments/files/19935004/test1_enc.xls)

**Comment by cool-firer** [comment]
ok then, any plan to support XLS encryption?

**Comment by zurmokeeper** [comment]
@cool-firer Yes, leave this issue is a reminder to do this function, but this is a little difficult, the biggest problem is not to find the encrypted document, can only rely on guessing, xls encryption is not a simple decryption of the reverse, I have spent a lot of time, but also not yet completed.

**Comment by cool-firer** [comment]
thanks
